source-http-file/main.go

```diff
@@ -20,10 +20,18 @@ type credentials struct {
 	Password string `json:"password"`
 }
 
+type headers struct {
+	Items []struct {
```

Comment:
Maybe call this `Extra` or something? Or just remove the extra level of nesting if you don't anticipate wanting to add `/headers/foo` properties for specific other headers?